cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/proto/ReportGenerationInfo.java

```diff
@@ -16,27 +16,74 @@
 
 package co.cask.cdap.report.proto;
 
+import javax.annotation.Nullable;
+
 /**
  * Represents the information of a report generation in an HTTP response.
  */
 public class ReportGenerationInfo {
+  private final String name;
+  @Nullable
+  private final String description;
   private final long created;
+  @Nullable
+  private final Long expiry;
   private final ReportStatus status;
-  private final String request;
+  @Nullable
+  private final String error;
```

Comment:
there is a lot of duplicate fields in `ReportGenerationInfo` and `ReportStatusInfo` that can be abstracted into a class.